### PR TITLE
added extension: filestorage

### PR DESCRIPTION
**Pull request by luckyj5**
Why this PR? 
SCK-Otel (https://github.com/splunk/sck-otel/tree/main/charts/sck-otel) is currently using opentelemetry-collector-contrib repo image, with this PR adding filestorage extension, we can use "splunk-otel-collector" repo image  with SCK-Otel as well for faster turnaround and lighter binary.

### CODE REVIEW

**Comment by dmitryax** [comment]
@luckyj5 please take a look into the failing tests

**Comment by emaderer** [comment]
@luckyj5  can you please add some description / context to this PR?

**Comment by luckyj5** [comment]
> @luckyj5 please take a look into the failing tests
Thanks @dmitryax. just updated the PR for failing tests.

**Comment by luckyj5** [comment]
> @luckyj5 can you please add some description / context to this PR?
Thanks @emaderer. Updated the description with the same. Please let me know if any questions.

File: docs/components.md [hunk 1]
```diff
@@ -17,9 +17,9 @@ removed prior to the 1.0 release.
...
-| [sapm](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/sapmreceiver)                         | [resource](https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/resourceprocessor)                           |                                                                                                                      |                                                                                                                              |
+| [sapm](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/sapmreceiver)                         | [resource](https://github.com/open-telemetry/opentelemetry-collector/tree/main/processor/resourceprocessor)                           |                                                                                                                      | [filestorage](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/storage/filestorage)     |                                                                                                                      |
```
Comment (rmfitzpatrick) [inline docs/components.md:22]: I believe this should be `file_storage` and it's in alpha: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/extension/storage/filestorage

Comment (luckyj5) [reply docs/components.md]: Thanks @rmfitzpatrick! Updated

**Review by rmfitzpatrick — APPROVED** [review]

**Pull request merged**